src/app/file-browser/terra-frontend-storage.service.ts

```diff
@@ -149,35 +151,69 @@ export class TerraFrontendStorageService extends TerraBaseService
         return item;
     }
 
-    public listFiles():Observable<TerraStorageObjectList>
+    public getMetadata( key: string ): Observable<TerraImageMetadata>
     {
-        return new Observable((observer:Observer<TerraStorageObjectList>) =>
+        if ( this._metadataCache.hasOwnProperty( key ) )
         {
-            if(this._storageInitialized)
-            {
-                observer.next(this._storageObjectList);
-            }
-            else
-            {
-                this.initStorageList();
+            return Observable.from( [this._metadataCache[key]] );
+        }
+
+        this.setAuthorization();
+        let request = this.mapRequest(
+            this.http.get(
+                "/rest/storage/frontend/file/metadata?key=" + key,
```

Comment:
man übergibt ja schon quasi "/rest/storage/frontend/file" im constructor... heißt müsstest einfach in der ganzen datei nur "this.url + '/metadata?key=' + key" machen